File: packages/whisker-local-store/src/ios/LocalStoreModule.swift
// Phase L-3 — `whisker-local-store` ModuleDefinition (iOS).
//
// Replaces the pre-L-3 `@WhiskerModule("WhiskerLocalStore")`-
// annotated `WhiskerLocalStoreImpl`. A view-less DSL module:
// `definition()` has no `View(...)` block, just module-level
// `Function`s. The SwiftPM codegen plugin discovers the
// `WhiskerModule` subclass, emits a `@_cdecl` dispatch shim, and
// registers it via `whisker_bridge_register_module_dispatch(
// "WhiskerLocalStore", shim)` — so
// `whisker::platform_module::invoke("WhiskerLocalStore", ...)`
// from Rust routes into these handlers.
//
// The storage logic lives in `LocalStore.swift`.

import WhiskerModuleApi

public final class LocalStoreModule: WhiskerModule {
    public override func definition() -> ModuleDefinition {
        ModuleDefinition {
            Name("WhiskerLocalStore")

            // save(key, value) -> Bool
            Function("save") { (key: String, value: String) -> Bool in
                LocalStore.save(key, value)
            }
            // load(key) -> String? | nil  (Rust lifts nil into Option::None)
            Function("load") { (key: String) -> String? in
                LocalStore.load(key)
            }
            // remove(key) -> Void  (collapses to .null → ())
            Function("remove") { (key: String) in
                LocalStore.remove(key)
            }
        }
    }
}

File: packages/whisker-local-store/src/android/LocalStoreModule.kt
// Phase L-3 — `whisker-local-store` ModuleDefinition (Android).
//
// Replaces the pre-L-3 `@WhiskerModule("WhiskerLocalStore")`-
// annotated `WhiskerLocalStoreImpl`. A view-less DSL module:
// `definition()` has no `View(...)` block, just module-level
// `Function`s. The KSP processor discovers the `WhiskerModule`
// subclass and calls `registerWithLynx()`, which registers the
// functions with `WhiskerModuleRegistry` under the `Name(...)` —
// so `whisker::platform_module::invoke("WhiskerLocalStore", ...)`
// from Rust routes into these handlers.
//
// The storage logic lives in `LocalStore.kt`.

package rs.whisker.modules.localstore

import rs.whisker.runtime.ModuleDefinition
import rs.whisker.runtime.WhiskerModule

class LocalStoreModule : WhiskerModule() {
    override fun definition() = ModuleDefinition {
        Name("WhiskerLocalStore")

        // save(key, value) -> Bool
        Function("save") { key: String, value: String ->
            LocalStore.save(key, value)
        }
        // load(key) -> Str | Null  (Rust lifts Null into Option::None)
        Function("load") { key: String ->
            LocalStore.load(key)
        }
        // remove(key) -> Null  (Unit return collapses to Null → ())
        Function("remove") { key: String ->
            LocalStore.remove(key)
        }
    }
}
